pycheribuild/config/compilation_targets.py

```diff
@@ -716,11 +729,16 @@ def toolchain_targets(cls, target: "CrossCompileTarget", config: "CheriConfig"):
         return ["morello-llvm-native"]
 
     @classmethod
-    def essential_compiler_and_linker_flags_impl(cls, *args, xtarget, **kwargs):
-        result = super().essential_compiler_and_linker_flags_impl(*args, xtarget=xtarget, **kwargs)
+    def essential_compiler_and_linker_flags_impl(cls, instance: "CheriBSDTargetInfo", *args, xtarget, **kwargs):
+        result = super().essential_compiler_and_linker_flags_impl(instance, *args, xtarget=xtarget, **kwargs)
+        version = instance.cheribsd_version()
+        if version >= 20220511:
```

Comment:
This does default to the old flags if there is no sys/param.h, but I think that is ok.